testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestGrantRevoke.java

```diff
@@ -129,6 +132,27 @@ public void testGrantRevoke()
                 .hasMessageContaining(format("Access Denied: Cannot select from table default.%s", tableName));
     }
 
+    @Test(groups = {AUTHORIZATION, PROFILE_SPECIFIC_TESTS})
+    public void testGrantRevokeWithGrantOption()
+    {
+        onTrino().executeQuery("CREATE ROLE role1 IN hive");
+        onTrino().executeQuery("CREATE ROLE role2 IN hive");
+        onTrino().executeQuery("GRANT role1 TO USER charlie IN hive");
+
+        // test GRANT WITH GRANT OPTION
+        aliceExecutor.executeQuery(format("GRANT SELECT ON %s TO bob WITH GRANT OPTION", tableName));
+        bobExecutor.executeQuery(format("GRANT SELECT ON %s TO ROLE role1", tableName));
+        assertThat(charlieExecutor.executeQuery(format("SELECT * FROM %s", tableName))).hasNoRows();
+
+        // test REVOKE WITH GRANT OPTION
+        aliceExecutor.executeQuery(format("REVOKE GRANT OPTION FOR SELECT ON %s FROM bob", tableName));
+        assertQueryFailure(() -> bobExecutor.executeQuery(format("GRANT SELECT ON %s TO ROLE role1 ", tableName)))
+                .hasMessageContaining(format("Access Denied: Cannot grant privilege SELECT on table default.%s", tableName));
+        // TODO (https://github.com/trinodb/trino/issues/4455) Should revoke only GRANT OPTION
+        assertQueryFailure(() -> bobExecutor.executeQuery(format("SELECT * FROM %s", tableName)))
+                .hasMessageContaining(format("Access Denied: Cannot select from table default.%s", tableName));
```

Comment:
Please also test if charlie still has access.

Reply:
Charlie would still have access. CASCADE is not supported in Hive.